builtin/providers/aws/cloudfront_distribution_configuration_structure.go

```diff
@@ -940,3 +974,28 @@ func flattenActiveTrustedSigners(ats *cloudfront.ActiveTrustedSigners) flatmap.M
 	m["items"] = s
 	return flatmap.Flatten(m)
 }
+
+// fullOK does a proper type-switch on an interface value and returns true
+// if the underlying type contains a zero value. float is ommitted as it is
+// currently not a data type in Terraform and definitely not in CloudFront.
+func fullOK(v interface{}) bool {
```

Comment:
This function seems pretty heavy handed... it only seems to be used with attributes that are strings, is there a situation where a simple check like so is insufficient? 

```go
if v, ok := m["attribute"]; ok && v.(string) != "" {
  [...]
}
```

I get that it checks types, but those attributes should all be `string`s right?